File: snackdown2016/1a.cpp
#include <iostream>
#include <ctime>

using namespace std;

int check(int a[100000],int n) {
    for (int i=0;i<n-2;i++) {
        if (a[i]==a[i+1] && a[i+1]==a[i+2])
            return 1;
    }
    return 0;
}
int main() {
    clock_t ti;
    int t,p=0;
    cin>>t;
    int arr[100000];
    while (t) {
        int n;
        cin>>n;
        ti = clock();
        int c[100000];
        for (int i=0;i<n;i++)
            cin>>c[i];
        if (check(c,n))
            arr[p++] = 1;
        else
            arr[p++] = 0;
        t--;
        ti = clock() - ti;
        cout<<((float)ti)/CLOCKS_PER_SEC<<endl;
    }
    for(int i=0;i<p;i++) {
        if (arr[i])
            cout<<"Yes"<<endl;
        else
            cout<<"No"<<endl;
    }
}

File: snackdown2016/1a.py
t = input()
ls = []
def check(a):
    for i in range(n-2):
        if a[i]==a[i+1]==a[i+2]:
            return 1
    return 0

while t:
    n = input()
    c = [int(i) for i in raw_input().split()]
    if check(c):
        ls.append("Yes")
    else:
        ls.append("No")
    t-=1

for i in ls:
    print i